service/src/main/java/com/ge/predix/acs/monitoring/DecisionCacheHealthIndicator.java

```diff
@@ -0,0 +1,59 @@
+package com.ge.predix.acs.monitoring;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.actuate.autoconfigure.ConditionalOnEnabledHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.context.annotation.Profile;
+import org.springframework.data.redis.connection.RedisConnection;
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.data.redis.core.RedisConnectionUtils;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+@Component
+@Profile({ "cloud-redis", "redis" })
+@ConditionalOnEnabledHealthIndicator("decision.cache")
```

Comment:
Yea that's probably better.. changed to using `UP` instead as the overall status if either caching or the Redis health check is disabled via environment variables.